pkg/network/setup/podnic.go

```diff
@@ -57,7 +58,12 @@ func newPhase2PodNIC(vmi *v1.VirtualMachineInstance, network *v1.Network, iface
 		return nil, err
 	}
 
-	ifaceLink, err := link.DiscoverByNetwork(podnic.handler, podnic.vmi.Spec.Networks, *podnic.vmiSpecNetwork, vmi.Status.Interfaces)
+	ifaceStatus := vmispec.LookupInterfaceStatusByName(vmi.Status.Interfaces, network.Name)
+	if ifaceStatus == nil {
+		return nil, fmt.Errorf("could not find interface status for %s", network.Name)
+	}
+
+	ifaceLink, err := podnic.handler.LinkByName(ifaceStatus.PodInterfaceName)
```

Comment:
Previously `newPhase2PodNIC` would failed on **non** LinkNotFound errors to allow a state where the hotplugged network's interface is not present in the pod yet https://github.com/kubevirt/kubevirt/commit/1c5cf683a1dbf6a8cb1432fbc40c957b6d1f3e9b.

Dont we need preserve that logic to avoid the VM from failing in above case?

Reply:
I think your suggestion makes sense.
@EdDev WDYT?

Reply:
I would be worried that this is not raising an error in the e2e or unit tests.
In the setup in which we do not perform in-place hotplug, it is expected for the interface not to appear in the pod.

Before you fix anything, I would go and check what happens in that scenario and why we do not see it manifested.

Reply:
Currently there are no negative unit-tests nor e2e for hotplug. 
On e2e, we test the positive scenario, where Multus work properly and the hotplugged interface present on time.

One way to verify this e2e is making Multus hang on attaching the hotplug interface (the CNI ADD call).
Long time ago I had to do something similar, I used this CNI https://github.com/fabiand/cni-hold-protoype 
to make Multus hang on the hotplug operation.
Basically its necessary to chain this CNI to the NAD that being used for the hotplug.